packages/reddit-helpers/src/lib/reddit.ts

```diff
@@ -34,6 +41,38 @@ export async function fetchRedditPosts(subreddit: string, limit = 30) {
 	return result.map((response) => handleResponse(subreddit, response));
 }
 
+/**
+ * Fetches a Reddit post from a specified subreddit.
+ * @param subreddit - The name of the subreddit this post belongs to.
+ * @param key - The key of the post to fetch.
+ * @returns A promise that resolves to the {@link RedditResponse} object contained the fetched post.
+ */
+export async function fetchRedditPost(subreddit: string, key: string) {
+	const cacheKey = `${subreddit}/${key}`;
+
+	const existing = cache.get(cacheKey);
+	if (!isNullish(existing)) return ok(existing);
+
+	const result = await getRequest<RedditResponse[]>(`r/${subreddit}/comments/${key}.json`);
+
+	return result
+		.mapInto((value) => {
+			const parsedPostResponse = parsePostResponse(value);
+			if (parsedPostResponse) return ok(parsedPostResponse);
+			return err(new RedditParseException('Failed to find a post in the response', subreddit, key));
+		})
```

Comment:
If `parsePostResponse` is refactored to return an `Option`:
```suggestion
		.mapInto((value) => parsePostResponse(value).match({
			some: (parsed) => ok(parsed),
			none: () => new RedditParseException('Failed to find a post in the response', subreddit, key)
		}))
```